<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>CryptoTool</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>0</y>
      <width>800</width>
      <height>600</height>
     </rect>
    </property>
    <property name="tabPosition">
     <enum>QTabWidget::TabPosition::North</enum>
    </property>
    <property name="tabShape">
     <enum>QTabWidget::TabShape::Rounded</enum>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_sm2">
     <attribute name="title">
      <string>SM2</string>
     </attribute>
     <widget class="QPushButton" name="pushButton_gen_sm2_keypair">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>40</y>
        <width>100</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>生成密钥对</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit_sm2_pubkey">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>10</y>
        <width>231</width>
        <height>91</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit_sm2_prikey">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>110</y>
        <width>231</width>
        <height>91</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_sm2_prikey_sign">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>90</y>
        <width>100</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>私钥签名</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit_sm2_sign_data">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>230</y>
        <width>231</width>
        <height>71</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit_sm2_sign_value">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>330</y>
        <width>231</width>
        <height>71</height>
       </rect>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_sm2_pubkey_verify">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>140</y>
        <width>100</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>公钥验签</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit_sm2_verify_result">
      <property name="geometry">
       <rect>
        <x>220</x>
        <y>430</y>
        <width>231</width>
        <height>71</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>40</y>
        <width>58</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>公钥</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>140</y>
        <width>58</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>私钥</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>240</y>
        <width>58</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>签名原文</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>360</y>
        <width>58</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>签名值</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>160</x>
        <y>460</y>
        <width>58</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>验签结果</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>20</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>加密原文</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit_sm2_encrypt_plain">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>10</y>
        <width>231</width>
        <height>91</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>130</y>
        <width>51</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>加密结果</string>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit_sm2_encrypt_result">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>110</y>
        <width>231</width>
        <height>91</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit_sm2_decrypt_result">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>330</y>
        <width>231</width>
        <height>71</height>
       </rect>
      </property>
     </widget>
     <widget class="QTextEdit" name="textEdit_sm2_decrypt_plain">
      <property name="geometry">
       <rect>
        <x>530</x>
        <y>230</y>
        <width>231</width>
        <height>71</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>360</y>
        <width>58</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>解密结果</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>470</x>
        <y>240</y>
        <width>58</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>解密原文</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_sm2_prikey_decrypt">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>240</y>
        <width>100</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>私钥解密</string>
      </property>
     </widget>
     <widget class="QPushButton" name="pushButton_sm2_pubkey_encrypt">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>190</y>
        <width>100</width>
        <height>32</height>
       </rect>
      </property>
      <property name="text">
       <string>公钥加密</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_sm3">
     <attribute name="title">
      <string>SM3</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="tab_sm4">
     <attribute name="title">
      <string>SM4</string>
     </attribute>
    </widget>
    <widget class="QWidget" name="tab_rsa">
     <attribute name="title">
      <string>RSA</string>
     </attribute>
    </widget>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
